packages/components/src/Actions/Action/Action.component.js

```diff
@@ -8,6 +8,52 @@ import CircularProgress from '../../CircularProgress';
 import Icon from '../../Icon';
 import getPropsFrom from '../../utils/getPropsFrom';
 
+const LEFT = 'left';
+const RIGHT = 'right';
+
+function getIcon({ icon, iconTransform, inProgress }) {
+	if (inProgress) {
+		return (<CircularProgress size="small" />);
+	}
+
+	if (icon) {
+		return (<Icon name={icon} transform={iconTransform} />);
+	}
+
+	return null;
+}
+getIcon.propTypes = {
+	icon: PropTypes.string,
+	iconTransform: PropTypes.string,
+	inProgress: PropTypes.bool,
+};
+
+function getLabel({ hideLabel, label }) {
+	return hideLabel ? null : <span>{label}</span>;
+}
+getLabel.propTypes = {
+	label: PropTypes.string,
+	hideLabel: PropTypes.bool,
+};
+
+function adjustContentPlacement(icon, label, iconPosition) {
+	if (iconPosition === RIGHT) {
+		return [label, icon];
+	}
+	return [icon, label];
+}
+
+function getContent(props) {
+	const icon = getIcon(props);
+	const label = getLabel(props);
```

Comment:
Why do you define constants when the functions can be called on the return statement?